### PR TITLE
RSA9h update Auth.CreateTokenRequest

### PR DESCRIPTION
**Pull request by gernest**

### CODE REVIEW

File: ably/auth_test.go [hunk 1]
```diff
@@ -604,26 +604,75 @@ func TestAuth_ClientID(t *testing.T) {
...
+	t.Run("RSA9c must generate a unique 16+ character nonce", func(ts *testing.T) {
+		req, err := client.Auth.CreateTokenRequest(params, opts)
+		if err != nil {
+			ts.Fatalf("CreateTokenRequest()=%v", err)
+		}
+		if len(req.Nonce) < 16 {
+			ts.Fatalf("want len(nonce)>=16; got %d", len(req.Nonce))
+		}
+	})
+	t.Run("generate a signed request", func(ts *testing.T) {
```
Comment (ORBAT) [inline ably/auth_test.go]: Is this `RSA9a` or `RSA9g`? Also, should we check for HMAC validity as well?

**Comment by gernest** [comment]
ping @paddybyers

**Comment by paddybyers** [comment]
I'd like to see also a test in here for RSA9i which in turn references RSA8b.

File: ably/auth_test.go [hunk 2]
```diff
@@ -604,26 +604,75 @@ func TestAuth_ClientID(t *testing.T) {
...
-		t.Fatalf("CreateTokenRequest()=%v", err)
-	}
-	if len(req.Nonce) < 16 {
-		t.Fatalf("want len(nonce)>=16; got %d", len(req.Nonce))
-	}
-	if req.Mac == "" {
-		t.Fatalf("want mac to be not empty")
-	}
+	t.Run("RSA9h", func(ts *testing.T) {
+		ts.Run("paramaters are optional", func(ts *testing.T) {
```
Comment (paddybyers) [inline ably/auth_test.go]: typo `parameters`

**Comment by ORBAT** [comment]
Good stuff. Thanks!

**Pull request merged**